src/Apps/Authentication/Components/AuthenticationInlineDialog.tsx

```diff
@@ -20,17 +20,23 @@ const AuthenticationInlineDialogContents: FC = () => {
     match: { location },
   } = useRouter()
 
-  // Errors might come back from 3rd party authentication
-  // as a string in an `error` query param, so display them if present.
+  // Errors might come back from 3rd party authentication so display them if present.
   useEffect(() => {
-    if (!location.query.error) return
+    if (!location.query.error_code) return
 
-    sendToast({
-      message: location.query.error,
-      variant: "error",
-      ttl: Infinity,
-    })
-  }, [location.query.error, sendToast])
+    const message = (
+      ERROR_CODES[location.query.error_code] || ERROR_CODES.UNKNOWN
+    )
+      // Resolve templates
+      .replace("{email}", location.query.email || "—")
```

Comment:
This substitution essentially still allows arbitrary injection into the message. Would it be OK to just modify the messages to be more generic (like `this email address` instead of `the email address {email}`)?

Reply:
The thing is, I've found that to be useful in the past — knowing what email address is actually associated, because I have a different one I use for Facebook than most other things, for instance.

I'll just port this to TS and pass the message through Sharify.

Reply:
Does that help the vulnerability though?

Reply:
Yeah, just pass the whole message from the server. Can't alter it or inject anything that way.